plugins/follow_me/follow_me_impl.cpp

```diff
@@ -0,0 +1,213 @@
+#include "follow_me_impl.h"
+#include "device_impl.h"
+#include "global_include.h"
+#include "px4_custom_mode.h"
+#include <functional>
+#include <cmath>
+
+namespace dronecore {
+
+FollowMeImpl::FollowMeImpl() :
+    PluginImplBase()
+{
+    _start = std::chrono::system_clock::now();
+    _estimatation_capabilities = 0;
+}
+
+FollowMeImpl::~FollowMeImpl()
+{
+}
+
+void FollowMeImpl::init()
+{
+    using namespace std::placeholders; // for `_1`
+
+    _parent->register_mavlink_message_handler(
+        MAVLINK_MSG_ID_HEARTBEAT,
+        std::bind(&FollowMeImpl::process_heartbeat, this, _1), static_cast<void *>(this));
+}
+
+void FollowMeImpl::deinit()
+{
+    _parent->unregister_all_mavlink_message_handlers(this);
+}
+
+void FollowMeImpl::timeout_occurred()
+{
+    // FIXME: This is HARD-coded as of now.
+    // Ideally GPS info should come from a platform-specific framwork.
+
+    // update current location coordinates
+    _motion_report.lat_int += 10;
```

Comment:
It would be nice to have an example that does a box or a circle.

Reply:
I will figure out the coordinates for bounding box and then add.